### DISCUSSION TITLE
Flashing stm32 after AI Deck

### DISCUSSION THREAD

**Discussion by Jool99** [bitcraze/discussions#1183] [Q&A]
Hi!

When we flash the crazyflie with our own app and then flash the AI deck with the instructions in the [wifi video streamer example](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/), everything works as we want.

But when flashing the stm32 again, the gap8 app is no longer started when starting the crazyflie.

So every time we want to update the cf2 firmware, we would have to:
- Flash stm32.
- Remove flow deck (flashing gets stuck at 6% otherwise).
- Flash gap8.
- Attach flow deck.

Is there a better way to do this?

Thanks in advance!

**Comment by gemenerik** [comment]
I can reproduce OTA flashing getting stuck with flow deck connected, will look into it.
I haven't had problems with flashing the STM32 after the AI-deck, however. My gap8 app remains intact. Are you flashing the same firmware both times?

I think for now it might be easiest to just flash your gap8 with a jtag programmer if you have one, so you do not have to remove the flow deck each time.

**Comment by Jool99** [comment]
Thanks for your answer!

We flashed a different STM32 firmware when the gap8 "broke". But reflashing the gap8 fixed it.

As for flashing with JTAG, we are running macOS on computers that do not support nested virtualization, so we cannot access USB from docker.

Is it possible to first build with docker and then flash with JTAG from outside docker?

**Comment by gemenerik** [comment]
> Is it possible to first build with docker and then flash with JTAG from outside docker?

That should be possible.
To compile your gap8 app:

```
docker run --rm -v ${PWD}:/module --device /dev/ttyUSB0 --privileged -P bitcraze/aideck tools/build/make-example [example-directory] all
```

Then with your native gap_sdk install; `make flash`
I'm not sure how much of the gap_sdk you will need to install. Theoretically you would only need the flashing component (I think Gapy), but for sure you should be able to avoid having to install nntool and the autotiler in this install.

Note that this overwrites the OTA bootloader so you would have to reflash it if you want to use it again later.

---

Flashing getting stuck is related to https://github.com/bitcraze/aideck-gap8-examples/issues/105

**Comment by Jool99** [comment]
Thanks for ur answer!

I have been able to flash OTA without the wifi streaming example stopping working a few times now but sometimes it does. Will write again if im able to reproduce it consistently.
